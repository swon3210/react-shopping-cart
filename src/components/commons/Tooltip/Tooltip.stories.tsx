import Tooltip, { Props } from './Tooltip';
import Button from '../Button/Button';

export default {
  title: 'Components/Commons/Tooltip',
  component: Tooltip,
  argTypes: {},
};

const Template = (args: Props) => <Tooltip {...args}>상품을 장바구니에 담았습니다</Tooltip>;

export const Default = Template.bind({});

(Default as any).args = {
  button: <Button size="SM">장바구니 가기</Button>,
};
