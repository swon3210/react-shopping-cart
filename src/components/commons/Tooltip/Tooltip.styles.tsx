import styled from 'styled-components';

export const Tooltip = styled.div`
  position: relative;
  width: 260px;
  padding: 24px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background: #ffffff;
  border: 2px solid #2ac1bc;
  &:after,
  &:before {
    top: 100%;
    left: 50%;
    border: solid transparent;
    content: '';
    height: 0;
    width: 0;
    position: absolute;
    pointer-events: none;
  }

  &:after {
    border-color: rgba(255, 255, 255, 0);
    border-top-color: #ffffff;
    border-width: 12px;
    margin-left: -12px;
  }

  &:before {
    border-color: rgba(42, 193, 188, 0);
    border-top-color: #2ac1bc;
    border-width: 15px;
    margin-left: -15px;
  }
`;

export const ToolTipText = styled.span`
  color: ${({ theme }) => theme.TEXT_COLOR};
  margin-bottom: 18px;
`;
export const ToolTipButtonWrapper = styled.div``;
