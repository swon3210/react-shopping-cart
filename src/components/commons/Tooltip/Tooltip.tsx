import { useEffect } from 'react';
import * as Styled from './Tooltip.styles';

export interface Props {
  children: React.ReactNode;
  button?: React.ReactNode;
  setTooltipShown?: React.Dispatch<React.SetStateAction<boolean>>;
  timeOut: number;
}

const Tooltip = ({ children, button, setTooltipShown, timeOut }: Props) => {
  useEffect(() => {
    const id = setTimeout(() => {
      setTooltipShown?.(false);
    }, timeOut);

    return () => clearTimeout(id);
  }, [setTooltipShown, timeOut]);

  return (
    <Styled.Tooltip>
      <Styled.ToolTipText>{children}</Styled.ToolTipText>
      <Styled.ToolTipButtonWrapper>{button}</Styled.ToolTipButtonWrapper>
    </Styled.Tooltip>
  );
};

export default Tooltip;
