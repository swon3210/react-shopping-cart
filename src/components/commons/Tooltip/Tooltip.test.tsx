import Tooltip from './Tooltip';
import { render } from '@testing-library/react';

describe('Tooltip Component', () => {
  it('Tooltip Snapshot', () => {
    const tooltipUtil = render(<Tooltip timeOut={0}>TEST</Tooltip>);
    expect(tooltipUtil.asFragment()).toMatchSnapshot();
  });
});
