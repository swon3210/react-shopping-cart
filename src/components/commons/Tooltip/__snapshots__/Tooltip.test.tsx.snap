// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Component Tooltip Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa kVccnQ"
  >
    <span
      class="sc-gsDJrp fikjTz"
    >
      TEST
    </span>
    <div
      class="sc-dkPtyc"
    />
  </div>
</DocumentFragment>
`;
